### PR TITLE
No need to del elem when config.CleanWindow>0 at method:Set

### PR DESCRIPTION
**Pull request by zhangyaping-CN**

### CODE REVIEW

**Review by cristaloleg — COMMENTED** [review]
SGTM, maybe a test?

File: shard.go [hunk 1]
```diff
@@ -129,8 +130,10 @@ func (s *cacheShard) set(key string, hashedKey uint64, entry []byte) error {
 		}
 	}
 
-	if oldestEntry, err := s.entries.Peek(); err == nil {
-		s.onEvict(oldestEntry, currentTimestamp, s.removeOldestEntry)
+	if !s.cleanEnabled {
```
Comment (janisz) [inline shard.go:133]: Could you add a test for this?

**Review by janisz — CHANGES_REQUESTED** [review]

File: shard.go (hunk 1 above)
Comment (zhangyaping-CN) [reply shard.go:133]: Thank you for your reply, I added testing and then added `if !s.cleanEnabled` when Append

**Review by cristaloleg — COMMENTED** [review]
SGTM

**Review by janisz — APPROVED** [review]

**Pull request merged**